### PR TITLE
[glib] Add –fstack-protector option

### LINKED ISSUE #4642: [glib] Failed to build
```shell
[136/1212] Linking target glib/libglib-2.0.so.0.6700.0
FAILED: glib/libglib-2.0.so.0.6700.0 
cc  -o glib/libglib-2.0.so.0.6700.0 glib/libglib-2.0.so.0.6700.0.p/deprecated_gallocator.c.o glib/libglib-2.0.so.0.6700.0.p/deprecated_gcache.c.o glib/libglib-2.0.so.0.6700.0.p/deprecated_gcompletion.c.o glib/libglib-2.0.so.0.6700.0.p/gpattern.c.o gl/usr/local/lib64/gcc/x86_64-cros-linux-gnu/10.2.0/../../../../x86_64-cros-linux-gnu/bin/ld: subprojects/sysprof/src/libsysprof-capture/libsysprof-capture-4.a(sysprof-collector.c.o): in function `sysprof_collector_get.part.0':
sysprof-collector.c:(.text+0x104): undefined reference to `__stack_chk_guard'
/usr/local/lib64/gcc/x86_64-cros-linux-gnu/10.2.0/../../../../x86_64-cros-linux-gnu/bin/ld: sysprof-collector.c:(.text+0x2a3): undefined reference to `__stack_chk_guard'
/usr/local/lib64/gcc/x86_64-cros-linux-gnu/10.2.0/../../../../x86_64-cros-linux-gnu/bin/ld: subprojects/sysprof/src/libsysprof-capture/libsysprof-capture-4.a(sysprof-collector.c.o):sysprof-collector.c:(.text+0x5ad): more undefined references to `__stack_chk_guard' follow
collect2: error: ld returned 1 exit status
[141/1212] Compiling C object glib/tests/array-test.p/array-test.c.o
ninja: build stopped: subcommand failed.
glib failed to install: `ninja -C build` exited with 1[136/1212] Linking target glib/libglib-2.0.so.0.6700.0
FAILED: glib/libglib-2.0.so.0.6700.0 
cc  -o glib/libglib-2.0.so.0.6700.0 glib/libglib-2.0.so.0.6700.0.p/deprecated_gallocator.c.o glib/libglib-2.0.so.0.6700.0.p/deprecated_gcache.c.o glib/libglib-2.0.so.0.6700.0.p/deprecated_gcompletion.c.o glib/libglib-2.0.so.0.6700.0.p/gpattern.c.o gl/usr/local/lib64/gcc/x86_64-cros-linux-gnu/10.2.0/../../../../x86_64-cros-linux-gnu/bin/ld: subprojects/sysprof/src/libsysprof-capture/libsysprof-capture-4.a(sysprof-collector.c.o): in function `sysprof_collector_get.part.0':
sysprof-collector.c:(.text+0x104): undefined reference to `__stack_chk_guard'
/usr/local/lib64/gcc/x86_64-cros-linux-gnu/10.2.0/../../../../x86_64-cros-linux-gnu/bin/ld: sysprof-collector.c:(.text+0x2a3): undefined reference to `__stack_chk_guard'
/usr/local/lib64/gcc/x86_64-cros-linux-gnu/10.2.0/../../../../x86_64-cros-linux-gnu/bin/ld: subprojects/sysprof/src/libsysprof-capture/libsysprof-capture-4.a(sysprof-collector.c.o):sysprof-collector.c:(.text+0x5ad): more undefined references to `__stack_chk_guard' follow
collect2: error: ld returned 1 exit status
[141/1212] Compiling C object glib/tests/array-test.p/array-test.c.o
ninja: build stopped: subcommand failed.
glib failed to install: `ninja -C build` exited with 1
```

### PR DESCRIPTION
**Pull request by supechicken666**
Fixes #4642

### CODE REVIEW

File: packages/glib.rb [hunk 1]
```diff
@@ -15,7 +15,7 @@ class Glib < Package
   depends_on 'util_linux'
   depends_on 'six'
   
-  ENV['CFLAGS'] = "-fno-stack-protector"
+  ENV['CFLAGS'] = "-fstack-protector"
```
Comment (uberhacker) [inline packages/glib.rb:18]: Isn't the default to enable stack protection?  @satmandu: Why did you need to add this option initially?

**Comment by satmandu** [comment]
I was having  a compilation failure with the earlier version...

**Comment by satmandu** [comment]
I can confirm that ```-fstack-protector``` also solves my compilation issue.

Using ```-fstack-protector``` is the better approach, obviously!

File: packages/glib.rb (hunk 1 above)
Comment (satmandu) [reply packages/glib.rb:18]: Either ```ENV['CFLAGS'] = "-fno-stack-protector"``` or ```ENV['CFLAGS'] = "-fstack-protector"``` solves my compilation issue!

**Comment by uberhacker** [comment]
Thank you for the update @supechicken666!

**Pull request merged**